### ISSUE TITLE
Contact query example from documentation no working

### ISSUE THREAD

**Issue by adrien-ben** [dimforge/ncollide#256]
Hi,

It's my first time using ncollide. For now I'm going through the documentation, and try to get the samples running. [The contact query example](https://raw.githubusercontent.com/sebcrozet/ncollide/master/build/ncollide2d/examples/contact_query2d.rs) does not work for me. The first assert fails (the prenetration one).

Slightly offsetting the ball (to 1.01, 1.01) makes it work

**Comment by sebcrozet** [comment]
Hi!

Thank you for reporting this. This is a bug, likely due to a regression on the contact computation algorithm.

**Comment by adrien-ben** [comment]
Glad to help!

**Comment by dawsonc** [comment]
I've encountered this bug as well. I don't know if this additional information could help, but I've found that (oddly) visualizing the scene using kiss3d causes the distance query to work correctly.

With the main function from the documentation, I get `penetrating_depth = 0.0`:
```
fn main() {
    let cuboid = Cuboid::new(Vector3::new(1.0, 1.0, 1.0));
    let ball = Ball::new(1.0);
    let prediction = 1.0;

    let cuboid_pos = na::one();
    let ball_pos_penetrating = Isometry3::new(Vector3::new(1.0, 1.0, 1.0), na::zero());

    let ctct_penetrating = query::contact(
        &ball_pos_penetrating,
        &ball,
        &cuboid_pos,
        &cuboid,
        prediction,
    );

    let penetrating_depth = ctct_penetrating.unwrap().depth;

    println!("ctct_penetrating depth: {}", penetrating_depth);
}
```

Whereas when I add additional code that uses the `cuboid_pos` or `ball_pos_penetrating` transformations in the kiss3d scene, I get the correct distance (`0.9978057`):

```
fn main() {
    let cuboid = Cuboid::new(Vector3::new(1.0, 1.0, 1.0));
    let ball = Ball::new(1.0);
    let prediction = 1.0;

    let cuboid_pos = na::one();
    let ball_pos_penetrating = Isometry3::new(Vector3::new(1.0, 1.0, 1.0), na::zero());

    let ctct_penetrating = query::contact(
        &ball_pos_penetrating,
        &ball,
        &cuboid_pos,
        &cuboid,
        prediction,
    );

    let penetrating_depth = ctct_penetrating.unwrap().depth;

    println!("ctct_penetrating depth: {}", penetrating_depth);

    let mut window = Window::new("Kiss3d: cube");
    let mut c      = window.add_cube(2.0, 2.0, 2.0);
    let mut b      = window.add_sphere(1.0);

    c.set_color(1.0, 0.0, 0.0);
    c.set_local_transformation(cuboid_pos);
    b.set_color(0.0, 0.0, 1.0);
    b.set_local_transformation(ball_pos_penetrating);

    window.set_light(Light::StickToCamera);

    while window.render() {
        // loop until user quits
    }
}
```

Narrowing it down, the minimal working example is that you need to either `c.set_local_transformation(cuboid_pos);` or `b.set_local_transformation(ball_pos_penetrating);` to get it to work. I'm new to this, so I'm stumped, but maybe this clue will help someone else.

**Comment by enebo** [comment]
I plan on only using aabb calculations for my project (so I do not think? this particular problem will bite me) but the very first example for this project I tried was contact (because I plan on using it).  I spent 15 minutes trying to make sense of the penetrating example failure shown above to realize it must be a bug.  Then upon reading the user guide I see the very same example. 

I know how hard it is to maintain OSS projects so I am not complaining but you will get a lot of people false-starting and walking away on this library when your documented example does not run properly.  I do not think I have the skills to help (my math is very very rusty) but if this issue is open much longer you may want to change the example to a passing version.

I thought about git bisecting but I went to 0.18.x and it was still not working.

**Comment by jruiseco** [comment]
Any updated on this bug? Seems like a showstopper for basically anyone trying to use the library.

**Comment by sebcrozet** [comment]
@jruiseco Looks like this issue has been forgotten. Thank you for raising it again. This will be fixed by #366.
